src/Core/OrganizationFeatures/OrganizationUsers/AcceptOrgUserCommand.cs

```diff
@@ -0,0 +1,224 @@
+﻿using Bit.Core.Auth.Models.Business.Tokenables;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.OrganizationFeatures.OrganizationUsers.Interfaces;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Settings;
+using Bit.Core.Tokens;
+using Bit.Core.Utilities;
+using Microsoft.AspNetCore.DataProtection;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationUsers;
+
+public class AcceptOrgUserCommand : IAcceptOrgUserCommand
+{
+    private readonly IDataProtector _dataProtector;
+    private readonly IGlobalSettings _globalSettings;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+    private readonly IOrganizationRepository _organizationRepository;
+    private readonly IPolicyService _policyService;
+    private readonly IMailService _mailService;
+    private readonly IUserRepository _userRepository;
+    private readonly IDataProtectorTokenFactory<OrgUserInviteTokenable> _orgUserInviteTokenDataFactory;
+
+    public AcceptOrgUserCommand(
+        IDataProtectionProvider dataProtectionProvider,
+        IGlobalSettings globalSettings,
+        IOrganizationUserRepository organizationUserRepository,
+        IOrganizationRepository organizationRepository,
+        IPolicyService policyService,
+        IMailService mailService,
+        IUserRepository userRepository,
+        IDataProtectorTokenFactory<OrgUserInviteTokenable> orgUserInviteTokenDataFactory)
+    {
+
+        // TODO: remove data protector when old token validation removed
+        _dataProtector = dataProtectionProvider.CreateProtector("OrganizationServiceDataProtector");
```

Comment:
suggestion: We can reuse the `OrgUserInviteTokenable.DataProtectorPurpose` instead of the magic string